plugins/reach/lib/kReachManager.php

```diff
@@ -44,14 +44,13 @@ protected function getObjectType($eventObjectClassName)
 		return null;
 	}
 
-	private function addingEntryVendorTaskByObjectIds($entryId, $allowedCatalogItemIds, $profileId, $object, $autoRule = false)
+	private function addingEntryVendorTaskByObjectIds($vendorTaskObjectHandler,  $allowedCatalogItemIds, $profileId, $object, $autoRule = false)
 	{
+		$taskObjectId = $vendorTaskObjectHandler->getTaskObjectId($object);
 		$catalogItemIdsToAdd = array_unique($allowedCatalogItemIds);
-
-		//If both the entry and reach profile don't exist, there's no need to hit the loop
-		$vendorTaskObjectHandler = HandlerFactory::getHandlerById($entryId, $object);
-		$taskObject = $vendorTaskObjectHandler->retrieveObject($entryId);
+		$taskObject = $vendorTaskObjectHandler->retrieveObject($taskObjectId);
```

Comment:
additional actions required: 
for 
public static function retrieveObject($objectId): BaseObject;

currently this signature throws an error in case object is not found, which is a change to previous behavior and is more strict, in some places in the code that uses this function, we bypass the object not found and continue (we do not stop the flow) 

please change the signature of the function and make sure it aligns in all places that uses it